ambari-server/src/main/java/org/apache/ambari/server/controller/internal/ServiceGroupResourceProvider.java

```diff
@@ -322,7 +322,7 @@ private ServiceGroupRequest getRequest(Map<String, Object> properties) {
   }
 
   // Get services from the given set of requests.
-  protected Set<ServiceGroupResponse> getServiceGroups(Set<ServiceGroupRequest> requests)
+  public Set<ServiceGroupResponse> getServiceGroups(Set<ServiceGroupRequest> requests)
     throws AmbariException {
     Set<ServiceGroupResponse> response = new HashSet<>();
     for (ServiceGroupRequest request : requests) {
```

Comment:
I've seen another example making such methods public. The drawback with making it public is that it will bypass authentication. On the other hand, it makes sense to call the object oriented methods of resource managers internally instead of calling the tedious hashmap based ones. @rnettleton @adoroszlai what do you think?

Reply:
While I see the convenience of making these methods public for easy consumption by internal services, I am a little worried that this bypasses authentication.

Can you ask @rlevas to review this change, since I think he implemented the authentication support in this area of the Ambari server-side code?

Reply:
I am fine with this, but it seems like getServiceGroups should be moved to some internal business logic class rather than remain in this class that handles API calls.